src/utils/limiter.ts

```diff
@@ -29,7 +26,9 @@ ratesLimiter.on('error', error => {
   logger.warn(`ratesLimiter error`, error);
 });
 
+const unlimitedRequests = BLOCKFROST_REQUEST_CONCURRENCY === 0;
```

Comment:
Not sure if it is a reasonable to set this to unlimited, ever. There is always some limit on OS level and too many promises in Node.js can cause issues. Omitting unlimited option would also simplify the problem with  `BLOCKFROST_REQUEST_CONCURRENCY || 1` since we could set the constant as
```
export const BLOCKFROST_REQUEST_CONCURRENCY = Number(
  process.env.BLOCKFROST_REQUEST_CONCURRENCY || 500,
);
```